fatal error C1083: Cannot open include file: 'stddef.h': No such file or directory

```
C:\>dotnet --version
5.0.100-preview.5.20258.6
```

Repro steps:
```
git clone https://github.com/AaronRobinsonMSFT/DNNE
cd DNNE\test\ExportingAssembly
dotnet publish
```

Result:
```
C:\DNNE\test\ExportingAssembly>dotnet publish
Microsoft (R) Build Engine version 16.7.0-preview-20257-02+b861333fc for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  Generated exports written to 'C:\DNNE\test\ExportingAssembly/obj\Debug\net5.0\dnne/ExportingAssembly.g.c'.
  Building native export: "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.27.28826\bin\Hostx64\x64\cl.exe" /Od /LDd /TC /MT /GS /Zi /D DNNE_ASSEMBLY_NAME=ExportingAssembly /I "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.27.28826\include" /I "C:\Users\jkotas\.nuget\packages\dnne\1.0.0\tools\platform" /I "C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x64\5.0.0-preview.5.20258.3\runtimes\win-x64\native" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\shared" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\um" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\ucrt" "C:\DNNE\test\ExportingAssembly\obj\Debug\net5.0\dnne\ExportingAssembly.g.c" "C:\Users\jkotas\.nuget\packages\dnne\1.0.0\tools\platform\platform.c"  /link /DLL /LIBPATH:"C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.27.28826\lib\x64" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.18362.0\um\x64" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.18362.0\ucrt\x64" "C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x64\5.0.0-preview.5.20258.3\runtimes\win-x64\native\libnethost.lib" Advapi32.lib /IGNORE:4099 /out:"C:\DNNE\test\ExportingAssembly\obj\Debug\net5.0\dnne\bin\ExportingAssemblyNE.dll"
  Microsoft (R) C/C++ Optimizing Compiler Version 19.27.28826 for x64
  Copyright (C) Microsoft Corporation.  All rights reserved.

  ExportingAssembly.g.c
C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x64\5.0.0-preview.5.20258.3\runtimes\win-x64\native\nethost.h(8): fatal error C1083: Cannot open include file: 'stddef.h': No such file or directory [C:\DNNE\test\ExportingAssembly\ExportingAssembly.csproj]
  platform.c
C:\Users\jkotas\.nuget\packages\dnne\1.0.0\tools\platform\platform.c(20): fatal error C1083: Cannot open include file: 'stddef.h': No such file or directory [C:\DNNE\test\ExportingAssembly\ExportingAssembly.csproj]
  Generating Code...
C:\Users\jkotas\.nuget\packages\dnne\1.0.0\build\DNNE.targets(102,5): error MSB3073: The command ""C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.27.28826\bin\Hostx64\x64\cl.exe" /Od /LDd /TC /MT /GS /Zi /D DNNE_ASSEMBLY_NAME=ExportingAssembly /I "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.27.28826\include" /I "C:\Users\jkotas\.nuget\packages\dnne\1.0.0\tools\platform" /I "C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x64\5.0.0-preview.5.20258.3\runtimes\win-x64\native" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\shared" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\um" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\ucrt" "C:\DNNE\test\ExportingAssembly\obj\Debug\net5.0\dnne\ExportingAssembly.g.c" "C:\Users\jkotas\.nuget\packages\dnne\1.0.0\tools\platform\platform.c"  /link /DLL /LIBPATH:"C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.27.28826\lib\x64" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.18362.0\um\x64" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.18362.0\ucrt\x64" "C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x64\5.0.0-preview.5.20258.3\runtimes\win-x64\native\libnethost.lib" Advapi32.lib /IGNORE:4099 /out:"C:\DNNE\test\ExportingAssembly\obj\Debug\net5.0\dnne\bin\ExportingAssemblyNE.dll" " exited with code 2. [C:\DNNE\test\ExportingAssembly\ExportingAssembly.csproj]
```

--- a/readme.md
+++ b/readme.md
@@ -101,7 +101,7 @@ Failure to load the runtime or find an export results in the native library call
 
     ```xml
     <ItemGroup>
-      <PackageReference Include="DNNE" Version="1.0.0" />
+      <PackageReference Include="DNNE" Version="1.0.1" />
     </ItemGroup>
     ```
 

--- a/src/dnne-gen/dnne-gen.csproj
+++ b/src/dnne-gen/dnne-gen.csproj
@@ -12,7 +12,7 @@
 
   <PropertyGroup>
     <PackageId>DNNE</PackageId>
-    <Version>1.0.0</Version>
+    <Version>1.0.1</Version>
     <Authors>AaronRobinsonMSFT</Authors>
     <Owners>AaronRobinsonMSFT</Owners>
     <Description>Package used to generated native exports for .NET assemblies.</Description>

--- a/src/msbuild/DNNE.BuildTasks/Windows.cs
+++ b/src/msbuild/DNNE.BuildTasks/Windows.cs
@@ -22,6 +22,7 @@ using Microsoft.VisualStudio.Setup.Configuration;
 using Microsoft.Win32;
 
 using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Text;
 
@@ -36,23 +37,15 @@ namespace DNNE.BuildTasks
         {
             export.Report(MessageImportance.Low, $"Building for Windows");
 
-            WinSDK winSdkDir = g_WinSdk.Value;
+            WinSDK winSdk = g_WinSdk.Value;
             string vsInstall = g_VsInstallPath.Value;
             string vcToolDir = GetVCToolsRootDir(vsInstall);
-            export.Report(CreateCompileCommand.DevImportance, $"VS Install: {vsInstall}\nVC Tools: {vcToolDir}");
+            export.Report(CreateCompileCommand.DevImportance, $"VS Install: {vsInstall}\nVC Tools: {vcToolDir}\nWinSDK Version: {winSdk.Version}");
 
             bool isDebug = IsDebug(export.Configuration);
             bool is64Bit = Is64BitTarget(export.Architecture);
 
             var archDir = is64Bit ? "x64" : "x86";
-
-            // WinSDK inc and lib paths
-            var winIncDir = Path.Combine(winSdkDir.Root, "Include", winSdkDir.Version);
-            var sharedIncDir = Path.Combine(winIncDir, "shared");
-            var umIncDir = Path.Combine(winIncDir, "um");
-            var ucrtIncDir = Path.Combine(winIncDir, "ucrt");
-            var umLibDir = Path.Combine(winSdkDir.Root, "Lib", winSdkDir.Version, "um", archDir);
-            var ucrtLibDir = Path.Combine(winSdkDir.Root, "Lib", winSdkDir.Version, "ucrt", archDir);
 
             // VC inc and lib paths
             var vcIncDir = Path.Combine(vcToolDir, "include");
@@ -70,13 +63,25 @@ namespace DNNE.BuildTasks
             compilerFlags.Append($"/TC /MT /GS /Zi ");
             compilerFlags.Append($"/D DNNE_ASSEMBLY_NAME={export.AssemblyName} ");
             compilerFlags.Append($"/I \"{vcIncDir}\" /I \"{export.PlatformPath}\" /I \"{export.NetHostPath}\" ");
-            compilerFlags.Append($"/I \"{sharedIncDir}\" /I \"{umIncDir}\" /I \"{ucrtIncDir}\" ");
+
+            // Add WinSDK inc paths
+            foreach (var incPath in winSdk.IncPaths)
+            {
+                compilerFlags.Append($"/I \"{incPath}\" ");
+            }
+
             compilerFlags.Append($"\"{export.Source}\" \"{Path.Combine(export.PlatformPath, "platform.c")}\" ");
 
             // Set linker flags
             linkerFlags.Append($"/DLL ");
             linkerFlags.Append($"/LIBPATH:\"{libDir}\" ");
-            linkerFlags.Append($"/LIBPATH:\"{umLibDir}\" /LIBPATH:\"{ucrtLibDir}\" ");
+
+            // Add WinSDK lib paths
+            foreach (var libPath in winSdk.LibPaths)
+            {
+                linkerFlags.Append($"/LIBPATH:\"{Path.Combine(libPath, archDir)}\" ");
+            }
+
             linkerFlags.Append($"\"{Path.Combine(export.NetHostPath, "libnethost.lib")}\" Advapi32.lib ");
             linkerFlags.Append($"/IGNORE:4099 "); // libnethost.lib doesn't ship PDBs so linker warnings occur.
             linkerFlags.Append($"/out:\"{Path.Combine(export.OutputPath, export.OutputName)}\" ");
@@ -181,39 +186,67 @@ namespace DNNE.BuildTasks
 
         private class WinSDK
         {
-            public string Root;
             public string Version;
+            public IEnumerable<string> IncPaths;
+            public IEnumerable<string> LibPaths;
+        }
+
+        private class VersionDescendingOrder : IComparer<Version>
+        {
+            public int Compare(Version x, Version y)
+            {
+                return y.CompareTo(x);
+            }
         }
 
         private static WinSDK GetLatestWinSDK()
         {
-            string win10sdkRoot;
-            string latestSdkVersionStr = null;
-            var latestSdkVersion = new Version();
             using (var kits = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\Windows Kits\Installed Roots"))
             {
-                win10sdkRoot = (string)kits.GetValue("KitsRoot10");
+                string win10sdkRoot = (string)kits.GetValue("KitsRoot10");
 
-                // Find the latest version
+                // Sort the entries in descending order as
+                // to defer to the latest version.
+                var versions = new SortedList<Version, string>(new VersionDescendingOrder());
+
+                // Collect the possible SDK versions.
                 foreach (var verMaybe in kits.GetSubKeyNames())
                 {
-                    if (!Version.TryParse(verMaybe, out Version latestMaybe)
-                        || latestMaybe < latestSdkVersion)
+                    if (!Version.TryParse(verMaybe, out Version versionMaybe))
                     {
                         continue;
                     }
 
-                    latestSdkVersion = latestMaybe;
-                    latestSdkVersionStr = verMaybe;
+                    versions.Add(versionMaybe, verMaybe);
+                }
+
+                // Find the latest version of the SDK.
+                foreach (var tgtVerMaybe in versions)
+                {
+                    // WinSDK inc and lib paths
+                    var incDir = Path.Combine(win10sdkRoot, "Include", tgtVerMaybe.Value);
+                    var libDir = Path.Combine(win10sdkRoot, "Lib", tgtVerMaybe.Value);
+                    if (!Directory.Exists(incDir) || !Directory.Exists(libDir))
+                    {
+                        continue;
+                    }
+
+                    var sharedIncDir = Path.Combine(incDir, "shared");
+                    var umIncDir = Path.Combine(incDir, "um");
+                    var ucrtIncDir = Path.Combine(incDir, "ucrt");
+                    var umLibDir = Path.Combine(libDir, "um");
+                    var ucrtLibDir = Path.Combine(libDir, "ucrt");
+
+                    return new WinSDK()
+                    {
+                        Version = tgtVerMaybe.Value,
+                        IncPaths = new[] { sharedIncDir, umIncDir, ucrtIncDir },
+                        LibPaths = new[] { umLibDir, ucrtLibDir },
+                    };
                 }
             }
 
-            if (string.IsNullOrEmpty(latestSdkVersionStr) || string.IsNullOrEmpty(win10sdkRoot))
-            {
-                throw new Exception("Unknown Win10 SDK version found.");
-            }
-
-            return new WinSDK() { Root = win10sdkRoot, Version = latestSdkVersionStr };
+            throw new Exception("No Win10 SDK version found.");
         }
     }
 }

--- a/test/ExportingAssembly/ExportingAssembly.csproj
+++ b/test/ExportingAssembly/ExportingAssembly.csproj
@@ -7,7 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="DNNE" Version="1.0.0" />
+    <PackageReference Include="DNNE" Version="1.0.1" />
   </ItemGroup>
 
 </Project>

